controllers/persistentvolumeclaim_controller.go

```diff
@@ -127,3 +61,19 @@ func (r *PersistentVolumeClaimReconciler) SetupWithManager(mgr ctrl.Manager) err
 		For(&corev1.PersistentVolumeClaim{}).
 		Complete(r)
 }
+
+func removeTopoLVMFinalizer(pvc *corev1.PersistentVolumeClaim) ([]string, bool) {
+	removed := false
+	// Due to the bug #310, multiple TopoLVM finalizers can exist in `pvc.Finalizers`.
+	// So we need to delete all of them.
+	for i := 0; i < len(pvc.Finalizers); {
+		if pvc.Finalizers[i] == topolvm.PVCFinalizer {
+			copy(pvc.Finalizers[i:], pvc.Finalizers[i+1:])
+			pvc.Finalizers = pvc.Finalizers[:len(pvc.Finalizers)-1]
```

Comment:
nits: might be simplified like

```suggestion
			pvc.Finalizers = append(pvc.Finalizers[:i], pvc.Finalizers[i+1:]...)
```

Reply:
Thanks. I fixed it as you proposed.